From 7ad393b5d5c5af4e095ff0e6aae08a5054c236af Mon Sep 17 00:00:00 2001
From: Aubrey Li <aubrey.li@intel.com>
Date: Sat, 10 May 2014 13:14:39 -0500
Subject: [PATCH 095/152] FIXME: Keep local APIC enabled during suspend for
 system wakeup

Power button on ASUS-T100 is supposed to wake the system up from
suspend sleep mode. However, the button is wired to the GPIO and
the interrupt is delivered to the processor via local APIC. So
the local APIC need to be enabled before we found another wakeup
capable device.

Signed-off-by: Aubrey Li <aubrey.li@intel.com>
---
 arch/x86/kernel/apic/apic.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/kernel/apic/apic.c b/arch/x86/kernel/apic/apic.c
index 7f26c9a..85e5c6d 100644
--- a/arch/x86/kernel/apic/apic.c
+++ b/arch/x86/kernel/apic/apic.c
@@ -2426,9 +2426,11 @@ static void apic_pm_activate(void)
 
 static int __init init_lapic_sysfs(void)
 {
+#if 0
 	/* XXX: remove suspend/resume procs if !apic_pm_state.active? */
 	if (cpu_has_apic)
 		register_syscore_ops(&lapic_syscore_ops);
+#endif
 
 	return 0;
 }
-- 
1.7.10.4

